### PR TITLE
Simplified http Regex

### LINKED ISSUE #241: Replace applyURL's http regex with a simpler version
Right now, [this regex expression](https://github.com/NIAEFEUP/nijobs-be/blob/6992a1a8cd7ec8ba249d97245054ed81645a24f4/src/models/modelUtils.js#L29) is unreadable and too complicated. The idea was to only allow valid URLs, but that's overkill in terms of security.

Instead, we can replace it with a simple RE that checks if the URL begins with http or https.

### PR DESCRIPTION
**Pull request by CarlosMealha**
Hey guys, I think it's everything okay with this regex, but if you have any suggestion feel free to give it

Closes #241

### CODE REVIEW

**Comment by BrunoRosendo** [comment]
@CarlosMealha Did you test the behaviour with React like it was asked in the issue? If so, could you elaborate on how that went?

**Comment by DoStini** [comment]
rebase

**Comment by BrunoRosendo** [comment]
https://github.com/NIAEFEUP/nijobs-fe/pull/249#discussion_r939547747

From this discussion, I think we should add a bit more complexity and verify if there's a dot as well. So, something like http(s)://[CHARS].[CHARS]

**Comment by CarlosMealha** [comment]
> [NIAEFEUP/nijobs-fe#249 (comment)](https://github.com/NIAEFEUP/nijobs-fe/pull/249#discussion_r939547747)
> 
> From this discussion, I think we should add a bit more complexity and verify if there's a dot as well. So, something like http(s)://[CHARS].[CHARS]

Okay, I will change it and give you my feedback

**Comment by CarlosMealha** [comment]
So I tested this regex with different props and it looks everything okay, the page didn't changed.

**Comment by Naapperas** [comment]
Once this and https://github.com/NIAEFEUP/nijobs-fe/pull/249 are merged, we can move this regex to [nijobs-commos](https://github.com/NIAEFEUP/nijobs-commons)

**Comment by Naapperas** [comment]
@CarlosMealha update the regex so we can merge this PR

**Comment by CarlosMealha** [comment]
> @CarlosMealha update the regex so we can merge this PR

I already updated the regex, do you want me to do anything else?

**Comment by BrunoRosendo** [comment]
> [NIAEFEUP/nijobs-fe#249 (comment)](https://github.com/NIAEFEUP/nijobs-fe/pull/249#discussion_r939547747)
> 
> From this discussion, I think we should add a bit more complexity and verify if there's a dot as well. So, something like http(s)://[CHARS].[CHARS]

You're missing the change we talked about in this comment. Adding .S+ probably does the trick :)
We should also add a test case for that: the success case already works so add a test where a URL like https://invalid should fail

**Comment by Naapperas** [comment]
On another note, once #245 gets merged, rebase so the audit run does not fail, then we can merge this

**Comment by CarlosMealha** [comment]
> On another note, once #245 gets merged, rebase so the audit run does not fail, then we can merge this

Okay guys, I'll do it

**Comment by CarlosMealha** [comment]
I changed the regex and I'm not 100% sure if this is what you expected:
image

I tested the expression online and it works, but my test is not passing, and I don't know why
image
image

**Comment by Naapperas** [comment]
> I changed the regex and I'm not 100% sure if this is what you expected: image
> 
> Also tested the expression online and it works, but it's not passing, and I don't know why image image

That's almost it: you are just missing the `//` part of the URL. Other than that, I don't think URLs with multiple successive `.` are valid, you just need to remove the `+` in front of the `\.` and that should be it.

**Comment by BrunoRosendo** [comment]
I didn't test it but the offer_params property should be applyURL instead of apply

File: src/models/modelUtils.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const validImageURL = (val) => {
 };
 
 export const validApplyURL = (val) => {
-    const httpRegex = /^https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*)?$/;
+    const httpRegex = /(^https?:\/\/)\S+\.\S+$/;
```
Comment (Naapperas) [inline src/models/modelUtils.js:29]:
```suggestion
    const httpRegex = /^https?:\/\/\S+\.\S+$/;
```
I don't see how a group is needed here but I might be missing something, if that's the case just mark this as resolved.

**Review by Naapperas — CHANGES_REQUESTED** [review]

File: src/models/modelUtils.js (hunk 1 above)
Comment (CarlosMealha) [reply src/models/modelUtils.js:29]: I just grouped to organize it

Comment (BrunoRosendo) [reply src/models/modelUtils.js:29]: @CarlosMealha Just remove those brackets and then we can merge this

**Review by BrunoRosendo — APPROVED** [review]

**Review by Naapperas — APPROVED** [review]

**Pull request merged**